webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/repository/ldap/LDAPRuleRepository.scala

```diff
@@ -147,7 +147,7 @@ class WoLDAPRuleRepository(
     })
   }
 
-  def delete(id:RuleId, modId: ModificationId, actor:EventActor, reason:Option[String]) : IOResult[DeleteRuleDiff] = {
+  private[this] def deleteRule(id:RuleId, modId: ModificationId, actor:EventActor, reason:Option[String]) : IOResult[DeleteRuleDiff] = {
```

Comment:
Should not be private and call internalDelete with boolean at false

Reply:
Should be renamed InternalDelete + a boolean named 'allowSystemDeletion'

Reply:
below line 

ZIO.when(oldCr.isSystem)

ZIO.when(!allowSystemDeletion && oldCr.isSystem)